Rakefile

```diff
@@ -1,6 +1,25 @@
 require "bundler/gem_tasks"
 require "rspec/core/rake_task"
 
+desc "Run RSpec code examples (skip postgres required ones)"
 RSpec::Core::RakeTask.new(:spec)
 
-task :default => :spec
+# In CI, as part of the .rspec_ci, load a helper that sets the configuration
+# setting to allow the postgres specs to run.
+desc "Run RSpec code examples (assumes ci dependencies)"
+RSpec::Core::RakeTask.new("spec:ci") do |t|
+  t.rspec_opts = "--options #{File.expand_path(".rspec_ci", __dir__)}"
+end
+
+desc "Run RSpec code examples (with local postgres dependencies)"
+RSpec::Core::RakeTask.new("spec:dev") do |t|
+  # Load the PostgresRunner helper to facilitate a clean postgres environment
+  # for testing locally (not necessary for CI), and enables the postgres test
```

Comment:
Even though it's not necessary for CI, this seems to over complicate the code (and making local runs "different").  Can we just run it in CI anyway presuming it's relatively inexpensive?

Reply:
So there is a couple of things here, but I did explain a lot of the architecture of this in the original PR here:

https://github.com/ManageIQ/manageiq-gems-pending/pull/385

So I won't fully repeat that here (and I honestly forget a lot of my decisions without using that to cheat), but the main take aways were:

- A travis box is ephemeral, so it is fine it we just nuke the entire DB, but on a local machine, a developers DB(s) for `vmdb_development` would be trashed every time this is run locally
- When trying to do TDD with this code, having a way to be able to validate you haven't borked anything locally (instead of waiting for Travis) is really nice.

* * *

That all said, I also agree that basically adding an "`init` script written in Ruby" (see `spec/postgres_runner_helper.rb `) is basically a lot of code to accomplish this, for like 2 tests...

They are arguably important tests, but depending on how much we do or do not move away from this backup strategy might determine if it is still of value.


P.S.  This is also still WIP, so definitely things are still subject to change a little more yet.